monix-execution/shared/src/main/scala/monix/execution/FutureUtils.scala

```diff
@@ -161,6 +172,10 @@ object FutureUtils {
     p.future
   }
 
+  /** Creates a future that never completes. */
+  def never[A]: Future[A] =
+    Promise[A].future
```

Comment:
Since I got notified since I chimed in about the MiMa thing thought I'll share some insight here
 too, feel free to tell me to go away if you'd prefer that :-)

I would say 👎, since this can create memory leaks in the form of callbacks registered on this future, which will never trigger, however still will be registered and kept in memory. 

Explanations: http://viktorklang.com/blog/Futures-in-Scala-2.12-part-6.html 
Scala 2.12 solution: https://github.com/scala/scala/blob/2.12.x/src/library/scala/concurrent/Future.scala#L561

Sneaky little edge case, isn't it